File: week2/atakantekatan/150. Evaluate Reverse Polish Notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = collections.deque()
        for token in tokens:
            if token == "+":
                n2 = stack.pop()
                n1 = stack.pop()
                OP = n1 + n2
                stack.append(OP)
            elif token == "-":
                n2 = stack.pop()
                n1 = stack.pop()
                OP = n1 - n2
                stack.append(OP)
            elif token == "*":
                n2 = stack.pop()
                n1 = stack.pop()
                OP = n1 * n2
                stack.append(OP)
            elif token == "/":
                n2 = stack.pop()
                n1 = stack.pop()
                OP = int(n1 / n2)
                stack.append(OP)
            else:
                stack.append(int(token))
        return stack.pop()

File: week2/Eren-Taskiran/150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack<>();
        String operator="*-/+";
        for(String token : tokens){
            if(token.length()==1 && operator.indexOf(token.charAt(0)) >= 0){
                int second = stack.pop();
                int first = stack.pop();
                int ans=0;
                if(token.equals("+")){
                    ans=first+second;
                }else if(token.equals("-")){
                    ans=first-second;
                }else if(token.equals("/")){
                    ans = first/second;
                }else if(token.equals("*")){
                    ans = first*second;
                }
                stack.push(ans);
            }else{
                stack.push(Integer.parseInt(token));
            }
        }
        return stack.pop();
    }
}